lite/dynamic_verifier.go

```diff
@@ -266,6 +268,50 @@ FOR_LOOP:
 	}
 }
 
+func (dv *DynamicVerifier) fetchAndverifyAllUpToHeight(h int64) (FullCommit, error) {
+	// Fetch latest full commit from trusted.
+	trustedFC, err := dv.trusted.LatestFullCommit(dv.chainID, 1, h)
+	if err != nil {
+		return FullCommit{}, err
+	}
+	// We already were at the requested height.
+	if trustedFC.Height() == h {
+		return trustedFC, nil
+	}
+	// fetch FullCommits height by height until we reach h
+	for heightToFetch := trustedFC.Height() + 1; heightToFetch <= h; heightToFetch++ {
+
+		nextFC, err := dv.source.LatestFullCommit(dv.chainID, heightToFetch, heightToFetch)
+		if err != nil {
+			return FullCommit{}, err
+		}
+		fmt.Println(nextFC.Height())
+		if nextFC.Height() != heightToFetch {
+			return FullCommit{}, lerr.ErrCommitNotFound()
+		}
+
+		if err := nextFC.ValidateFull(dv.chainID); err != nil {
+			return FullCommit{}, err
+		}
+		// Update height for height.
+		// TODO: do we want the lite client to store the intermediate commits,
+		//  or only the latest/last one?
+		if err := dv.verifyAndSave(trustedFC, nextFC); err != nil {
```

Comment:
I'm not sure about this but there are actually no reasons to save the intermediate full-commits here, right? 
@ebuchman @melekes

Reply:
How do we benefit from storing them? maybe only that if something fails, we'll start from the last saved commit, no?

Reply:
I think the only benefit would be that if the application constantly queries state of the past, we have more trusted (verified) full commits / signed headers stored locally which can be used as a starting point for this queries instead having to query them from `dv.source`. My intuition would be that storing the last header / full commit should be sufficient but this intuition is not based on any numbers/data.

Reply:
If we want to query past state, we need past Merkle roots (app hashes). I don't think we need the full commits though.

Reply:
Thanks Chris. Probably just storing the latest full commit should be enough. The full commit contains the AppHashes via the Header:
https://github.com/tendermint/tendermint/blob/15f621141dce76d992f6c8dcfbd4c522878b6108/lite/commit.go#L16-L20
https://github.com/tendermint/tendermint/blob/15f621141dce76d992f6c8dcfbd4c522878b6108/types/block.go#L689-L690
https://github.com/tendermint/tendermint/blob/15f621141dce76d992f6c8dcfbd4c522878b6108/types/block.go#L376

We could probably get rid of some (most?) of the fields here for the lite client. 

For now, storing only the latest FullCommit sounds like the most reasonable thing to do.